OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeployWithCharge.cs

```diff
@@ -217,8 +217,10 @@ void ITick.Tick(Actor self)
 						chargedToken = self.RevokeCondition(chargedToken);
 				}
 				else
+				{
 					if (chargedToken == Actor.InvalidConditionToken)
 						chargedToken = self.GrantCondition(Info.ChargedCondition);
+				}
```

Comment:
this was supposed to be allowed 🤔

Reply:
The rule wants:
```c#
if (chargeTick < (deployed ? deployedChargeThreshold : chargeThreshold))
{
	if (chargedToken != Actor.InvalidConditionToken)
		chargedToken = self.RevokeCondition(chargedToken);
}
else if (chargedToken == Actor.InvalidConditionToken)
	chargedToken = self.GrantCondition(Info.ChargedCondition);
```
but I felt keeping the visual symmetry was a better outcome:
```c#
if (chargeTick < (deployed ? deployedChargeThreshold : chargeThreshold))
{
	if (chargedToken != Actor.InvalidConditionToken)
		chargedToken = self.RevokeCondition(chargedToken);
}
else
{
	if (chargedToken == Actor.InvalidConditionToken)
		chargedToken = self.GrantCondition(Info.ChargedCondition);
}
```

Happy to take either though.